File: ocs_samples/library_samples/Java/ocs_sample_library_preview/src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/SdsInterpolationMode.java
/** SdsInterpolationMode.java
 * 
 */

package  com.github.osisoft.ocs_sample_library_preview.sds;

/**
 * SdsInterpolationMode 0-3
 */
public enum SdsInterpolationMode {

    Continuous(0),
    StepwiseContinuousLeading(1),
    StepwiseContinuousTrailing(2),
    Discrete(3);
    private final int SdsInterpolationMode;

    private SdsInterpolationMode(int id) {
        this.SdsInterpolationMode = id;
    }
    
    /**
     * gets the int value
     * @return
     */
    public int getValue() {
        return SdsInterpolationMode;
    }
}

File: ocs_samples/basic_samples/SDS/DotNet/SdsRestApiCore/SdsRestApiCore/SdsInterpolationMode.cs
// <copyright file="SdsInterpolationMode.cs" company="OSIsoft, LLC">
//
// </copyright>

using System;
using System.Collections.Generic;
using System.Text;

namespace SdsRestApiCore
{
    public enum SdsInterpolationMode
    {
        Default = Continuous,
        Continuous = 0,
        StepwiseContinuousLeading = 1,
        StepwiseContinuousTrailing = 2,
        Discrete = 3
    }
}
